src/pybind/mgr/mds_autoscaler/module.py

```diff
@@ -0,0 +1,139 @@
+
+import logging
+from typing import Optional, List
+from mgr_module import MgrModule
+from ceph.deployment.service_spec import ServiceSpec
+import orchestrator
+import copy
+
+log = logging.getLogger(__name__)
+
+
+class MDSAutoscaler(orchestrator.OrchestratorClientMixin, MgrModule):
+    """
+    MDS autoscaler.
+    """
+    def __init__(self, *args, **kwargs):
+        super(MDSAutoscaler, self).__init__(*args, **kwargs)
+        self.set_mgr(self)
+
+    def get_service(self, fs_name: str) -> List[orchestrator.ServiceDescription]:
+        service = f"mds.{fs_name}"
+        completion = self.describe_service(service_type='mds',
+                                           service_name=service,
+                                           refresh=True)
+        self._orchestrator_wait([completion])
+        orchestrator.raise_if_exception(completion)
+        return completion.result
+
+    def get_daemons(self, fs_name: str) -> List[orchestrator.DaemonDescription]:
+        service = f"mds.{fs_name}"
+        completion = self.list_daemons(service_name=service)
+        self._orchestrator_wait([completion])
+        orchestrator.raise_if_exception(completion)
+        return completion.result
+
+    def update_daemon_count(self, fs_name: str, abscount: int) -> ServiceSpec:
+        svclist = self.get_service(fs_name)
+        assert svclist is not None and len(svclist) > 0
+        svc = svclist[0]
+
+        # no need to update placement count if the current placement count is
+        # the required count
+        assert svc.spec.placement.count != abscount
+
+        ps = copy.deepcopy(svc.spec.placement)
+        ps.count = abscount
+        newspec = ServiceSpec(service_type=svc.spec.service_type,
+                              service_id=svc.spec.service_id,
+                              placement=ps)
+        return newspec
+
+    def get_required_standby_count(self, fs_map: dict, fs_name: str) -> int:
+        assert fs_map is not None
+        for fs in fs_map['filesystems']:
+            if fs['mdsmap']['fs_name'] == fs_name:
+                return fs['mdsmap'].get('standby_count_wanted')
+        assert False
+
+    def get_current_standby_count(self, fs_map: dict, fs_name: str, daemons: List[orchestrator.DaemonDescription]) -> int:
+        # standbys are not grouped by filesystems in fs_map
+        # available = standby_replay + standby_active
+        assert fs_map is not None
+        total = 0
+        daemon_names = {
+            d.name() for d in daemons
+        }
+        for sb in fs_map['standbys']:
+            full_name = "mds.{}".format(sb['name'])
+            if full_name in daemon_names:
+                total += 1
+        return total
+
+    def get_current_active_count(self, fs_map: dict, fs_name: str) -> int:
+        assert fs_map is not None
+        for fsys in fs_map['filesystems']:
+            if fsys.get('mdsmap').get('fs_name') == fs_name:
+                return len(fsys.get('mdsmap').get('up'))
```

Comment:
Since you're looking at the daemon names in `get_current_standby_count` to determine _which_ standbys belong to a file system, you need to do the same here. If the daemon was not spawned by the orchestrator, it should not be counted.

Reply:
fixed